### DISCUSSION TITLE
Convert a MT4 .ex4 to a MT5 EA

### DISCUSSION THREAD

**Discussion by bhairav23** [EA31337/EA31337#400] [Q&A] [answered]
i do not have the source code, only the .ex4 and I had requested the freelancer to lock it so it only works on a specific MT4 account. I need it now to be accessible to any MT5 account ideally and need the source code of it. is this something someone can help with?

**Comment by kenorb** [comment, marked as answer]
Having binary in .ex4 format it's not possible to edit it or read its source code. If it is yours EA and you lost access to the source code, the only way is to use decompiler which I don't have it.
I could only help you with conversion to MQL5 while having MQL4 source code using the existing libraries.
Otherwise, you'll have to write your EA from scratch.
